src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -1084,11 +1108,18 @@ function withContext<T>(
         return isTruncated ? factory.createKeywordTypeNode(SyntaxKind.AnyKeyword) : result;
     }
 
-    function tryGetReturnType(node: SignatureDeclaration): Type | undefined {
-        const signature = typeChecker.getSignatureFromDeclaration(node);
-        if (signature) {
-            return typeChecker.getReturnTypeOfSignature(signature);
-        }
+    function typePredicateToTypeNode(typePredicate: TypePredicate, enclosingDeclaration: Node, flags = NodeBuilderFlags.None): TypeNode | undefined {
+        let isTruncated = false;
+        const result = typePredicateToAutoImportableTypeNode(typeChecker, importAdder, typePredicate, enclosingDeclaration, scriptTarget, declarationEmitNodeBuilderFlags | flags, {
+            moduleResolverHost: program,
+            trackSymbol() {
+                return true;
+            },
+            reportTruncationError() {
+                isTruncated = true;
+            },
+        });

```

Comment:
This looks pretty much the same as `typeToTypeNode`; why is it that we need a second set of helpers for type predicates? Why can't `typeToAutoImportableTypeNode` "just work" for these cases?

Reply:
Because type predicates are neither types nor type nodes :cry:

Reply:
😭